src/goto-programs/goto_convert_side_effect.cpp

```diff
@@ -96,7 +96,7 @@ void goto_convertt::remove_assignment(
 
     exprt rhs;
 
-    const typet &op0_type=ns.follow(expr.op0().type());
+    const typet &op0_type = expr.op0().type();
```

Comment:
Note the test at line 108 would therefore have been unreachable prior to this change -- but evidently our tests did not reveal that. Add a test?

Reply:
This not only was dead code, it actually still is dead code: the front-end already introduces type casts (both the C and C++ front-ends take care of this). I have, in a second commit, removed the dead code and added a `PRECONDITION` in the `else` case.